tests/pytest_framework/functional_tests/profiling/test_with_valgrind.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2015-2018 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import pytest
+
+
+def process_valgrind_log(valgrind_log_path):
+    with open(valgrind_log_path, "r") as valgrind_log:
+        valgrind_content = valgrind_log.read()
+        assert "Invalid read" not in valgrind_content
+        assert "Invalid write" not in valgrind_content
+
+
+@pytest.mark.slow
```

Comment:
That would be so awesome if we had a `@valgrind` decorator instead, not sure how difficult to write such a decorator though (or even possible?).